app/services/supplier_chooser.rb

```diff
@@ -1,16 +1,86 @@
+  # | Supplier1            | Supplier1           | Supplier1       |
+  # | Supplier1            | Supplier2           | Supplier1       |
+  #
+  # | Supplier2            | nil                 | Supplier2       |
+  # | Supplier2            | Supplier1           | Supplier2       |
+  # | Supplier2            | Supplier2           | Supplier2       |
+
+  # UPDATING EXISTING RECORDS
+  # | Location+Date Owner | Existing Owner | Doorkeeper App Owner | Resulting Owner |
+  # |---------------------|----------------|----------------------|-----------------|
+  # | nil                 | nil            | nil                  | nil             |
+  # | nil                 | nil            | Supplier1            | Supplier1       |
+  # | nil                 | nil            | Supplier2            | Supplier2       |
+  #
+  # | nil                 | Supplier1      | nil                  | Supplier1       |
+  # | nil                 | Supplier1      | Supplier1            | Supplier1       |
+  # | nil                 | Supplier1      | Supplier2            | Supplier1       |
+  #
+  # | nil                 | Supplier2      | nil                  | Supplier2       |
+  # | nil                 | Supplier2      | Supplier1            | Supplier2       |
+  # | nil                 | Supplier2      | Supplier2            | Supplier2       |
+  #
+  # | Supplier1           | nil            | nil                  | Supplier1       |
+  # | Supplier1           | nil            | Supplier1            | Supplier1       |
+  # | Supplier1           | nil            | Supplier2            | Supplier1       |
+  #
+  # | Supplier1           | Supplier1      | nil                  | Supplier1       |
+  # | Supplier1           | Supplier1      | Supplier1            | Supplier1       |
+  # | Supplier1           | Supplier1      | Supplier2            | Supplier1       |
+  #
+  # | Supplier1           | Supplier2      | nil                  | Supplier1       |
+  # | Supplier1           | Supplier2      | Supplier1            | Supplier1       |
+  # | Supplier1           | Supplier2      | Supplier2            | Supplier1       |
+  #
+  # | Supplier2           | nil            | nil                  | Supplier2       |
+  # | Supplier2           | nil            | Supplier1            | Supplier2       |
+  # | Supplier2           | nil            | Supplier2            | Supplier2       |
+  #
+  # | Supplier2           | Supplier1      | nil                  | Supplier2       |
+  # | Supplier2           | Supplier1      | Supplier1            | Supplier2       |
+  # | Supplier2           | Supplier1      | Supplier2            | Supplier2       |
+  #
+  # | Supplier2           | Supplier2      | nil                  | Supplier2       |
+  # | Supplier2           | Supplier2      | Supplier1            | Supplier2       |
+  # | Supplier2           | Supplier2      | Supplier2            | Supplier2       |
```

Comment:
@martyn-w it would be great if you could talk people through your thinking with this logic, I'm not fully following the approach. Eventually it would be good to capture this truth table in the form of specs too - appreciate that wasn't the focus whilst burning the midnight oil but maybe mark with a TODO for the future.